src/ucp/proto/proto_single.inl

```diff
@@ -0,0 +1,123 @@
+/**
+ * Copyright (C) Mellanox Technologies Ltd. 2020.  ALL RIGHTS RESERVED.
+ *
+ * See file LICENSE for terms.
+ */
+
+#ifndef UCP_PROTO_SINGLE_INL_
+#define UCP_PROTO_SINGLE_INL_
+
+#include "proto_single.h"
+#include "proto_common.inl"
+
+#include <ucp/dt/datatype_iter.inl>
+
+
+static UCS_F_ALWAYS_INLINE ucs_status_t
+ucp_proto_am_bcopy_single_send(ucp_request_t *req, ucp_am_id_t am_id,
+                               ucp_lane_index_t lane,
+                               uct_pack_callback_t pack_func, void *pack_arg,
+                               size_t max_packed_size)
+{
+    ucp_ep_t *ep                       = req->send.ep;
+    const uct_iface_attr_t *iface_attr = ucp_ep_get_iface_attr(ep, lane);
+    ssize_t packed_size;
+    uint64_t *buffer;
+
+    if ((max_packed_size <= UCS_ALLOCA_MAX_SIZE) &&
```

Comment:
`max_packed_size <= ucs_min(UCS_ALLOCA_MAX_SIZE, iface_attr->cap.am.max_short)`?

Reply:
max_packed_size could be compile-time constant sometimes